### PR TITLE
TimeFileMaker: use accurate sum in sum_times

### LINKED ISSUE #18266: Test suite (sometimes?) fails on MacOS
And what's up with
```
logs//coq-makefile/timing.log
+ set -e
+ ./001-correct-diff-sorting-order/run.sh
+ set -e
++ dirname ./001-correct-diff-sorting-order/run.sh
+ cd ./001-correct-diff-sorting-order
+ /Users/runner/work/coq/coq/_install_ci/lib/coq-core/tools/make-both-time-files.py time-of-build-after.log.in time-of-build-before.log.in time-of-build-both-user.log
+ diff -u time-of-build-both-user.log.expected time-of-build-both-user.log
+ /Users/runner/work/coq/coq/_install_ci/lib/coq-core/tools/make-both-time-files.py --real time-of-build-after.log.in time-of-build-before.log.in time-of-build-both-real.log
+ diff -u time-of-build-both-real.log.expected time-of-build-both-real.log
--- time-of-build-both-real.log.expected	2023-11-04 07:12:35.000000000 +0000
+++ time-of-build-both-real.log	2023-11-04 07:35:50.000000000 +0000
@@ -1,6 +1,6 @@
     After |   Peak Mem | File Name                                      |    Before |   Peak Mem ||    Change || Change (mem) | % Change | % Change (mem)
 ---------------------------------------------------------------------------------------------------------------------------------------------------------
-20m46.07s | 3302508 ko | Total Time / Peak Mem                          | 23m06.30s | 3307052 ko || -2m20.23s ||     -4544 ko |  -10.11% |         -0.13%
+20m46.06s | 3302508 ko | Total Time / Peak Mem                          | 23m06.30s | 3307052 ko || -2m20.24s ||     -4544 ko |  -10.11% |         -0.13%
 ---------------------------------------------------------------------------------------------------------------------------------------------------------
  4m16.77s | 1617000 ko | Specific/X25519/C64/ladderstep                 |  5m16.83s | 1621500 ko || -1m00.06s ||     -4500 ko |  -18.95% |         -0.27%
  3m01.77s | 1589516 ko | Specific/solinas32_2e255m765_13limbs/femul     |  3m27.94s | 1656912 ko || -0m26.16s ||    -67396 ko |  -12.58% |         -4.06%
+ exit 1
==========> FAILURE <==========
    coq-makefile/timing/run.sh...Error!

FAILURES
    coq-makefile/timing/run.sh...Error!
make[1]: *** [report] Error 1
make: *** [all] Error 2
```
On macos test suite?  I'll try rerunning it to see if it's transient, but this seems concerning

_Originally posted by @JasonGross in https://github.com/coq/coq/issues/17832#issuecomment-1795658058_

### PR DESCRIPTION
**Pull request by SkySkimmer**
This should the tests with python 3.12 (sum changed behaviour on float list)

Fix https://github.com/coq/coq/issues/18266

### CODE REVIEW

**Comment by SkySkimmer** [comment]
I guess the closest float to `1246.07` is `1246.069999999999936335370875895023345947265625`, and improving accuracy of sum in python 3.12 changed the sum from `1246.07000000000016370904631912708282470703125` to that so the result of `int` (which truncates) changed.

File: tools/TimeFileMaker.py [hunk 1]
```diff
@@ -306,20 +307,20 @@ def from_seconds(seconds, signed=False):
     Converts a number of seconds into a string time.
     '''
     sign = ('-' if seconds < 0 else '+') if signed else ''
-    seconds = abs(seconds)
+    full_centiseconds=round(abs(seconds) * 100)
```
Comment (JasonGross) [inline tools/TimeFileMaker.py:310]: For consistency with surrounding code
```suggestion
    full_centiseconds = round(abs(seconds) * 100)
```

File: tools/TimeFileMaker.py [hunk 2]
```diff
@@ -306,20 +307,20 @@ def from_seconds(seconds, signed=False):
...
+    seconds = int(full_centiseconds) // 100
+    centiseconds = full_centiseconds - (seconds * 100)
```
Comment (JasonGross) [inline tools/TimeFileMaker.py:312]: Parentheses are not necessary, but I'm happy to have it either way
```suggestion
    centiseconds = full_centiseconds - seconds * 100
```

**Review by JasonGross — APPROVED** [review]
LGTM once CI passes; thanks for doing this!

**Comment by SkySkimmer** [comment]
It works

**Comment by JasonGross** [comment]
@coqbot merge now

**Comment by MSoegtropIMC** [comment]
I don't think that this PR fundamentally fixes the issue. The python team might decide to do the same to `math.fsum` they did to `sum`  - and I would say it is not unlikely that they do this. The only way to really fix this is to use a number system which is associative (for the numbers at hand). This means rationals, scaled integers or decimal floats - all of which are available in core python. But there is no point to over engineer this - we can fix it again if it breaks again. What would be a bit safer is a plain "sort" + "fold" btw - it is unlikely that someone changes fold such that it does things in an unexpected order.

**Pull request merged**